packages/flutter/lib/src/cassowary/constraint.dart

```diff
@@ -5,21 +5,46 @@
 import 'priority.dart';
 import 'expression.dart';
 
+/// A constraint is created by specifying a relationship between two
+/// expressions. The solver tries to satisfy this relationship after the
+/// constraint has been added to it at a set priority.
 enum Relation {
+  /// The relationship between the left and right hand sides of the expression
+  /// is `==`, (lhs == rhs).
   equalTo,
+  /// The relationship between the left and right hand sides of the expression
+  /// is `<=`, (lhs <= rhs).
   lessThanOrEqualTo,
+  /// The relationship between the left and right hand sides of the expression
+  /// is `>=`, (lhs => rhs).
   greaterThanOrEqualTo,
 }
 
+/// A constraint is a relationship between two expressions that the solver tries
+/// to hold true. In case of ambiguities, the solver will use priorities to
+/// determine constraint precedence. Once a constraint is added to the solver,
+/// this priority may not be changed.
 class Constraint {
+  /// Create a new constraint by specifying a single expression. This assumes
+  /// that the right hand side expression is the constant zero.
+  /// (`<expression> <relation> <0>`)
   Constraint(this.expression, this.relation);
 
+  /// The relationship between constraint expression and zero.
   final Relation relation;
 
+  /// The constraint expression.
   final Expression expression;
 
+  /// The constraint priority. The priority may only be modified when the
+  /// constraint is being created. Once it is added to the solver, modifications
+  /// to the constraint priority will have no effect on the how the solver
+  /// evaluates the constraint.
   double priority = Priority.required;
 
+  /// The operator `|` is overloaded as a convenience so that constraint
+  /// priorities can be specifed along with the constraint expression.
+  /// Usage: `ax + by + cx <= 0 | Priority.weak`.
```

Comment:
Put a black line before "Usage". Also maybe make that "For example:" rather than "Usage".